### ISSUE TITLE
drm-next 4.6 radeonkms failed

### ISSUE THREAD

**Issue by linuxcompitech** [FreeBSDDesktop/freebsd-base-graphics#87]
I installed FreeBSD 10.3 after this I built drm-next 4.6. When I restart the FreeBSD System with the new kernel, I get the following error message. Where am I going wrong?

```
KLD ng_ubt.ko: depends on kernel - not available or version mismatch
linker_load_file: Unsupported file type
```

lancs

**Comment by nomadlogic** [comment]
The work in this repository all tracks FreeBSD-CURRENT.  You are trying to load a kernel for FreeBSD-CURRENT on a 10.3 system which will not work.  You will have to build/install the "world" in addition to the kernel to test out the code in this repository.

You can follow the directions here to get you started with this process:
https://www.freebsd.org/doc/handbook/makeworld.html

I'll also update the wiki to make this explicitly clear.

**Comment by mattmacy** [comment]
That's not the problem causing this particular error.

 Any modules outside of /boot/kernel need to be rebuilt against the new kernel, which usually requires an updated world. That error means there was a version mismatch between the module and the running kernel.

**Comment by mattmacy** [comment]
@linuxcompitech where is your ng_ubt.ko?

**Comment by linuxcompitech** [comment]
@mattmacy Probably a FreeBSD-CURRENT - FreeBSD 10.3 conflict. I will check that with FreeBSD 12.0-Current.

**Comment by topcat001** [comment]
Hi, not sure if I should create a new thread for this, but I'm also trying to run freebsd-current (drm-next-4.6) on an older Optiplex 755 with a Radeon 2400XT (r600 chip). I have built the drm-next-4.6 kernel, userland, and the updated X server and utilities. I have observed two things:

If I run startx without preloading the radeonkms driver, then startx is "successful", but the screen goes blank (powered off). Logging in remotely shows Xorg is running with no errors in Xorg.0.log.

If I load the firmware using /boot/loader.conf (like I do on another identical system running 10.3):
`radeonkmsfw_RV610_me_load="YES"`
`radeonkmsfw_RV610_pfp_load="YES"`
`radeonkmsfw_R600_rlc_load="YES"`
`radeonkms_load="YES"`
then I get a panic during boot. I understand the radeon driver is untested. Let me know if I can help. Thanks!
